pandas/src/period.pyx

```diff
@@ -0,0 +1,1121 @@
+        """
+        Convert Period to desired frequency, either at the start or end of the
+        interval
+
+        Parameters
+        ----------
+        freq : string
+        how : {'E', 'S', 'end', 'start'}, default 'end'
+            Start or end of the timespan
+
+        Returns
+        -------
+        resampled : Period
+        """
+        from pandas.tseries.frequencies import get_freq_code as _gfc
+        how = _validate_end_alias(how)
+        base1, mult1 = _gfc(self.freq)
+        base2, mult2 = _gfc(freq)
+
+        if mult2 != 1:
+            raise ValueError('Only mult == 1 supported')
+
+        end = how == 'E'
+        new_ordinal = period_asfreq(self.ordinal, base1, base2, end)
+
+        return Period(ordinal=new_ordinal, freq=base2)
+
+    @property
+    def start_time(self):
+        return self.to_timestamp(how='S')
+
+    @property
+    def end_time(self):
+        if self.ordinal == tslib.iNaT:
+            ordinal = self.ordinal
+        else:
+            ordinal = (self + 1).start_time.value - 1
+        return Timestamp(ordinal)
+
+    def to_timestamp(self, freq=None, how='start', tz=None):
+        """
+        Return the Timestamp representation of the Period at the target
+        frequency at the specified end (how) of the Period
+
+        Parameters
+        ----------
+        freq : string or DateOffset, default is 'D' if self.freq is week or
+               longer and 'S' otherwise
+            Target frequency
+        how: str, default 'S' (start)
+            'S', 'E'. Can be aliased as case insensitive
+            'Start', 'Finish', 'Begin', 'End'
+
+        Returns
+        -------
+        Timestamp
+        """
+        from pandas.tseries import frequencies
+        from pandas.tseries.frequencies import get_freq_code as _gfc
+        how = _validate_end_alias(how)
+
+        if freq is None:
+            base, mult = _gfc(self.freq)
+            freq = frequencies.get_to_timestamp_base(base)
+
+        base, mult = _gfc(freq)
+        val = self.asfreq(freq, how)
+
+        dt64 = period_ordinal_to_dt64(val.ordinal, base)
+        return Timestamp(dt64, tz=tz)
+
+    year = _period_field_accessor('year', 0)
+    month = _period_field_accessor('month', 3)
+    day = _period_field_accessor('day', 4)
+    hour = _period_field_accessor('hour', 5)
+    minute = _period_field_accessor('minute', 6)
+    second = _period_field_accessor('second', 7)
+    weekofyear = _period_field_accessor('week', 8)
+    week = weekofyear
+    dayofweek = _period_field_accessor('dayofweek', 10)
+    weekday = dayofweek
+    dayofyear = _period_field_accessor('dayofyear', 9)
+    quarter = _period_field_accessor('quarter', 2)
+    qyear = _period_field_accessor('qyear', 1)
+
+    @classmethod
+    def now(cls, freq=None):
+        return Period(datetime.now(), freq=freq)
+
+    # HACK IT UP AND YOU BETTER FIX IT SOON
+    def __str__(self):
+        return self.__unicode__()
+
+    def __repr__(self):
```

Comment:
again in next PR, ``Period`` should inherit from ``PandasObject`` [here](https://github.com/pydata/pandas/blob/master/pandas/core/base.py)